data/modules/FlightLog/FlightLog.lua

```diff
@@ -341,51 +260,43 @@ function FlightLog:GetLogEntries(types, maximum, earliest_first)
 	end
 end
 
--- LOGGING
-
--- onLeaveSystem
-local AddSystemDepartureToLog = function (ship)
-	if not ship:IsPlayer() then return end
-
-	FlightLog.AddEntry( FlightLogEntry.System.New( Game.system.path, nil, Game.time, nil ) )
-end
-
--- onEnterSystem
-local AddSystemArrivalToLog = function (ship)
-	if not ship:IsPlayer() then return end
-
-	FlightLog.AddEntry( FlightLogEntry.System.New( Game.system.path, Game.time, nil, nil ) )
-end
-
--- onShipDocked
-local AddStationToLog = function (ship, station)
-	if not ship:IsPlayer() then return end
-
-	FlightLog.AddEntry( FlightLogEntry.Station.New( station.path, Game.time, Game.player:GetMoney(), nil ) )
-end
-
-function FlightLog.OrganizeEntries()
-
-	local function sortf( a, b )
-		return a.sort_date > b.sort_date
+local entryRecoveryLookup = {
+	["FlightLogEntry.Custom"] = FlightLogEntry.Custom.Unserialize,
+	["FlightLogEntry.System"] = FlightLogEntry.System.Unserialize,
+	["FlightLogEntry.Station"] = FlightLogEntry.Station.Unserialize
+}
+
+local function recoverEntry( entry, pathFixup, classLookup )
+	local class = classLookup( entry );
+	local unserializer = entryRecoveryLookup[class]
+	if unserializer == nil then
+		logWarning( "Flightlig unserialozer for " .. class .. " not found" )		
+	else
+		logInfo( "Found unserializer for " .. class )
 	end
-
-	table.sort( FlightLogData, sortf )
-
-	CollapseSystemEvents()
+	if unserializer == nil then return nil end
+	return unserializer( entry, pathFixup )
 end
 
--- LOADING AND SAVING
+--- @param loaded_data	table						The data to load
+--- @param pathFixup	func(table):SystemPath		Optional function to fixup invalid system paths
+--- @param classLookup  func(table):string			Lookup for classname.
+--- @return boolean									Did we succesfully read the data?
+function FlightLog.ParseSavedData(loaded_data, pathFixup, classLookup)
 
-local loaded_data
 
-local onGameStart = function ()
+	if loaded_data == nil then return false end
+
+	if loaded_data.Version == 1 then
```

Comment:
I mentioned this on IRC as well, but since we've done a savebump to v90, saves containing the version 1 flight log data structures are incompatible and thus this entire if-branch is redundant. Instead, the save recovery function for the flight log data will be used.

Reply:
So I think this and the below comment are kind of what I'm getting at.

There are two options I can see:
1) The flight-log code owns the flight log serialization, including recovery, which can be generalized as any system own's it's own code and specific recovery logic
2) The flight-log code only owns loading the latest** version of the code and recovery is separated into the new-game menu (or some other module), which handles recovery for everything.

I was also asking the same question of the UI rendering :-)

I'll comment more below as this ties into the next comment